---
title: "Retina ABCA4KO Treated vs ABCA4KO Untreated Diff Test"
author: "Maya Bhattiprolu, David McGaughey"
date: "6/1/23"
output: 
  html_notebook:
    toc: TRUE
    toc_float: TRUE
---


```{r}
library(tidyverse)
library(tximport)
library(DESeq2)

list.files(path = "/Users/bhattiprolum2/Desktop/MitraMouse/")
afile = tximport('/Users/bhattiprolum2/Desktop/MitraMouse/MF_10M/quant.sf', type = "salmon", txOut = TRUE)

#install.packages('data.table')
files <- list.files(path ='/Users/bhattiprolum2/Desktop/MitraMouse/', recursive = TRUE, pattern = 'quant.sf', full.names = TRUE)

anno <- read_tsv(files[1])
anno$Gene <-sapply(anno$Name,function(x)strsplit(x,"\\|")[[1]][6])
anno_tximport <-anno %>%
  dplyr::select(target_id = Name, Gene)

txi <- tximport(files, type = "salmon", tx2gene = anno_tximport)
txi.dseq2 <- data.frame(txi$counts)

samples <- str_extract(files, "MF_\\d+M")
colnames(txi.dseq2) <- samples
colnames(txi.dseq2)

mouse <- read.csv('/Users/bhattiprolum2/Desktop/mitra_mouse.csv')

colnames(mouse)[1] <- "Sample"

mouse_metadata <- colnames(txi.dseq2) %>% 
  enframe() %>% 
  left_join(., mouse %>% mutate(value = gsub("-","_",  Sample), by = c("value"))) %>%  
  separate(Sample.description, c("Genotype","Treatment"), sep = '- ') %>% 
  mutate(Treatment = tolower(Treatment), 
                Treatment=trimws(Treatment))
dds <- DESeqDataSetFromTximport(txi, mouse_metadata, design = ~Tissue)
DESeq2Table <- DESeq(dds, parallel = T)
vst <- varianceStabilizingTransformation(DESeq2Table)
colnames(vst) <- colData(dds)$value
mouse_metadata %>% DT::datatable()

mouse_metadata2 <- mouse_metadata %>%
  dplyr::select(Sample, Treatment, Genotype, Tissue) %>% 
  mutate(MisLabel = case_when(Sample == 'MF-10M' ~ "Retina?",
                              Sample == 'MF-6M' ~ "Retina?"))
```

# Diff test
retina ABCA4KO treated vs ABCA4KO untreated
```{r}
dds_against_treatment <- DESeqDataSetFromTximport(txi, mouse_metadata, design = ~Treatment)
colnames(dds_against_treatment) <- colData(dds)$value
dds_retina_ABCA4KO <- dds_against_treatment[,c('MF_11M', 'MF_9M', 'MF_5M', 'MF_7M')]

dds_retina_ABCA4KO$Treatment <- droplevels(dds_retina_ABCA4KO$Treatment)

dds_retina_ABCA4KO <-DESeq(dds_retina_ABCA4KO)

resABCA4KO <- results(dds_retina_ABCA4KO)

resABCA4KO_lfc <- lfcShrink(dds_retina_ABCA4KO, coef = "Treatment_untreated_vs_400.mg", type = "normal")

resABCA4KO_tibble <- resABCA4KO_lfc %>% 
  as_tibble(rownames = "Gene") 

resABCA4KO_tibble %>%
  arrange(desc(log2FoldChange))

resABCA4KO_tibble %>%
  arrange(padj)
```
## ma plot
```{r}
plotMA(resABCA4KO, ylim = c(-2,2))
plotMA(resABCA4KO_lfc, ylim = c(-2,2))
```
## volcano plot
```{r}
library(ggrepel)
volcano_ABCA4KO <- resABCA4KO %>%
  as_tibble(rownames = "Gene") %>%
  filter(!is.na(log2FoldChange)) 

volcano_ABCA4KO_filter1 <- volcano_ABCA4KO %>%
  filter(padj < 0.01, log1p(baseMean) > 4) %>%
  filter(log2FoldChange >1 | log2FoldChange < -1)

volcano_ABCA4KO_filter2 <- volcano_ABCA4KO %>%
  arrange(padj) %>%
  filter(padj < 0.01, log1p(baseMean) > 4) %>%
  filter(log2FoldChange >1) %>%
  head(5)

volcano_ABCA4KO_filter3 <- volcano_ABCA4KO %>%
  arrange(padj) %>%
  filter(padj < 0.01) %>%
  filter(log2FoldChange < -1)

volcano_ABCA4KO %>%
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = Gene)) +
  geom_point() +
  geom_point(data = volcano_ABCA4KO_filter1, color = "blue") +
  geom_label_repel(data = volcano_ABCA4KO_filter2) + 
  geom_label_repel(data = volcano_ABCA4KO_filter3)
  
```
## genes
```{r}
library(ggthemes)
retina_ABCA4KO_counts <- as.data.frame(assay(dds_retina_ABCA4KO))

retina_ABCA4KO_counts %>%
  as_tibble(rownames = "Gene") %>%
  filter(Gene %in% c('Trf', 'Rgr', 'C4b', 'Ly75', 'Pdlim3', 'Kif4', 'Nsl1', 'Tbx5')) %>%
  pivot_longer(-Gene) %>%
  left_join(mouse_metadata %>% dplyr::select(name, value, Genotype, Treatment, Tissue), by = c('name' = 'value')) %>%
  ggplot(aes(x = Treatment, y = log1p(value), color = name)) +
  geom_boxplot(color = 'black') +
  geom_point(size = 2) +
  facet_wrap(~Gene, scales = 'free') + 
  labs( y = 'log counts') +
  theme_dark()
```
## PCA

```{r}
ntop = 1000
vst_retinaKO <- vst[,c('MF_11M', 'MF_9M', 'MF_5M', 'MF_7M')]
scaled_counts_retinaKO <- assay(vst_retinaKO)

Pvars <- rowVars(scaled_counts_retinaKO)
select <- order(Pvars, decreasing = TRUE)[seq_len(min(ntop, 
                                                        length(Pvars)))]
PCA_retinaKO <- prcomp(t(scaled_counts_retinaKO[select, ]), scale = F)
percentVar <- round(100*PCA_retinaKO$sdev^2/sum(PCA_retinaKO$sdev^2),1)


dataGG = data.frame(PC_retinaKO1 = PCA_retinaKO$x[,1], PC_retinaKO2 = PCA_retinaKO$x[,2], 
                     PC_retinaKO3 = PCA_retinaKO$x[,3], PC_retinaKO4 = PCA_retinaKO$x[,4], 
                     tissue = colData(vst_retinaKO)$Tissue,
                     treatment=colData(vst_retinaKO)$Treatment,
                     genotype = colData(vst_retinaKO)$Genotype)


ggplot(dataGG, aes(PC_retinaKO1, PC_retinaKO2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + labs(title = "Retina ABCA4KO PCA 1/2")

ggplot(dataGG, aes(PC_retinaKO3, PC_retinaKO4, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC3: ",percentVar[3],"% variance")) +
  ylab(paste0("PC4: ",percentVar[4],"% variance")) + labs(title = "Retina ABCA4KO PCA 3/4")
```
## go enrichment
```{r}
#BiocManager::install('enrichplot')
#BiocManager::install('org.Mm.eg.db')
#BiocManager::install('clusterProfiler')

library(clusterProfiler)
library(enrichplot)
library(org.Mm.eg.db)

diff_genes <- resABCA4KO %>% 
  as_tibble(rownames = 'Gene') %>% 
  filter(padj < 0.01) %>%
  filter(abs(log2FoldChange) >1)


eg_diff_genes <- bitr(diff_genes$Gene, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")
eg_diff_genes <- diff_genes %>%
  left_join(., eg_diff_genes, by = c('Gene' = 'SYMBOL'))

# only keep genes that have a mean of >= 10 reads across all sample

expressed <- txi$counts[txi$counts %>% rowMeans() > 10,] %>% row.names()
eg_universe = bitr(resABCA4KO %>% 
                              as_tibble(rownames = 'Gene') %>% 
                              filter(Gene %in% expressed) %>% 
                              pull(Gene), fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")

ego <- enrichGO(gene          = eg_diff_genes$ENTREZID,
                universe      = eg_universe$ENTREZID,
                OrgDb         = org.Mm.eg.db,
                ont           = "all",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
                readable      = TRUE)

```

### dot plot
```{r}
p1 <- dotplot(ego, showCategory=30) + ggtitle("Dotplot for GO")
p1
```

### table
```{r}
ego <- setReadable(ego, OrgDb = org.Mm.eg.db, keyType = 'ENTREZID')
ego@result %>% DT::datatable()
```
### cnet plot
```{r, fig.width=6, fig.height=4}
geneList <- eg_diff_genes$log2FoldChange
names(geneList) <- eg_diff_genes$Gene
cnet <- cnetplot(ego, foldChange = geneList) + scale_color_viridis_c(name = 'log2(FoldChange)')
cnet
```

### Enriched GO Term Relationships
```{r, fig.width=8, fig.height=8}
ego_pair <- pairwise_termsim(ego) 
go_term_relationships <- emapplot(ego_pair,layout="kk") #+ ggtitle("Enriched GO Term Relationships")
go_term_relationships
```

# session info
```{r}
sessionInfo()
```

    
    