---
title: "Retina Untreated WT/ABCA4KO vs Treated ABCA4KO Diff Test"
author: "Maya Bhattiprolu, David McGaughey"
date: "6/1/23"
output: 
  html_notebook:
    toc: TRUE
    toc_float: TRUE
---


```{r}
library(tidyverse)
library(tximport)
library(DESeq2)

list.files(path = "/Users/bhattiprolum2/Desktop/MitraMouse/")
afile = tximport('/Users/bhattiprolum2/Desktop/MitraMouse/MF_10M/quant.sf', type = "salmon", txOut = TRUE)

#install.packages('data.table')
files <- list.files(path ='/Users/bhattiprolum2/Desktop/MitraMouse/', recursive = TRUE, pattern = 'quant.sf', full.names = TRUE)

anno <- read_tsv(files[1])
anno$Gene <-sapply(anno$Name,function(x)strsplit(x,"\\|")[[1]][6])
anno_tximport <-anno %>%
  dplyr::select(target_id = Name, Gene)

txi <- tximport(files, type = "salmon", tx2gene = anno_tximport)
txi.dseq2 <- data.frame(txi$counts)

samples <- str_extract(files, "MF_\\d+M")
colnames(txi.dseq2) <- samples
colnames(txi.dseq2)

mouse <- read.csv('/Users/bhattiprolum2/Desktop/mitra_mouse.csv')

colnames(mouse)[1] <- "Sample"

mouse_metadata <- colnames(txi.dseq2) %>% 
  enframe() %>% 
  left_join(., mouse %>% mutate(value = gsub("-","_",  Sample), by = c("value"))) %>%  
  separate(Sample.description, c("Genotype","Treatment"), sep = '- ') %>% 
  mutate(Treatment = tolower(Treatment), 
                Treatment=trimws(Treatment))
dds <- DESeqDataSetFromTximport(txi, mouse_metadata, design = ~Tissue)
DESeq2Table <- DESeq(dds, parallel = T)
vst <- varianceStabilizingTransformation(DESeq2Table)
colnames(vst) <- colData(dds)$value

mouse_metadata2 <- mouse_metadata %>%
  dplyr::select(Sample, Treatment, Genotype, Tissue) %>% 
  mutate(MisLabel = case_when(Sample == 'MF-10M' ~ "Retina?",
                              Sample == 'MF-6M' ~ "Retina?"))
mouse_metadata2

```

# Diff test
Retina untreated WT/ABCA4KO vs treated ABCA4KO
```{r}

dds_against_treatment2 <- DESeqDataSetFromTximport(txi, mouse_metadata, design = ~Treatment)
colnames(dds_against_treatment2) <- colData(dds)$value
dds_retina_treatment <- dds_against_treatment2[,c('MF_11M', 'MF_1M', 'MF_3M', 'MF_5M', 'MF_7M', 'MF_9M')]

dds_retina_treatment$Treatment <- droplevels(dds_retina_treatment$Treatment)

dds_retina_treatment <-DESeq(dds_retina_treatment)

res_treatment <- results(dds_retina_treatment)

res_treatment_lfc <- lfcShrink(dds_retina_treatment, coef = "Treatment_untreated_vs_400.mg", type = "normal")

res_treatment_tibble <- res_treatment_lfc %>% 
  as_tibble(rownames = "Gene") 

res_treatment_tibble %>%
  arrange(desc(log2FoldChange))

res_treatment_tibble %>%
  arrange(padj)


```


## ma plot
```{r}
plotMA(res_treatment, ylim = c(-2,2))
plotMA(res_treatment_lfc, ylim = c(-2,2))

```
## volcano
```{r}
library(ggrepel)

volcano_treatment<- res_treatment %>%
  as_tibble(rownames = "Gene") %>%
  filter(!is.na(log2FoldChange)) 


volcano_treatment_filter1 <- volcano_treatment %>%
  arrange(padj) %>%
  filter(padj < 0.01) %>%
  filter(log2FoldChange >1 | log2FoldChange < -1)

volcano_treatment_filter2 <-volcano_treatment %>%
  arrange(padj) %>%
  filter(padj < 0.01) %>%
  filter(log2FoldChange >1)

volcano_treatment_filter3 <- volcano_treatment %>%
  arrange(padj) %>%
  filter(padj < 0.01) %>%
  filter(log2FoldChange < -1)


volcano_treatment %>%
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = Gene)) +
  geom_point() +
  geom_point(data = volcano_treatment_filter1, color = "orange") +
  geom_label_repel(data = volcano_treatment_filter2) +
  geom_label_repel(data = volcano_treatment_filter3)

```
## genes
```{r}
library(ggthemes)
retina_treatment_counts <- as.data.frame(assay(dds_retina_treatment))

retina_treatment_counts %>%
  as.tibble(rownames = "Gene") %>%
  filter(Gene %in% c('Obscn', 'Nsl1', 'Rspo2', 'Sst', 'Fcgr3', 'Adgrl4', 'Ip6k3', 'Fam205a2')) %>%
  pivot_longer(-Gene) %>%
  left_join(mouse_metadata %>% dplyr::select(name, value, Genotype, Treatment, Tissue), by = c('name' = 'value')) %>%
  ggplot(aes(x = Treatment, y = log1p(value), color = name, shape = Genotype)) +
  geom_boxplot(color = "black") +
  geom_point() +
  facet_wrap(~Gene, scales = "free")+ 
  labs( y = 'log counts') +
  theme_bw()

```
## pca
```{r}
library(matrixStats)
ntop = 1000
vst_retinatreatment <- vst[,c('MF_11M', 'MF_1M', 'MF_3M', 'MF_5M', 'MF_7M', 'MF_9M')]
scaled_counts_retinatreatment <- assay(vst_retinatreatment)

Pvars <- rowVars(scaled_counts_retinatreatment)
select <- order(Pvars, decreasing = TRUE)[seq_len(min(ntop, 
                                                        length(Pvars)))]
PCA_retinatreatment <- prcomp(t(scaled_counts_retinatreatment[select, ]), scale = F)
percentVar <- round(100*PCA_retinatreatment$sdev^2/sum(PCA_retinatreatment$sdev^2),1)


dataGG = data.frame(PC_retinatreatment1 = PCA_retinatreatment$x[,1], PC_retinatreatment2 = PCA_retinatreatment$x[,2], 
                     PC_retinatreatment3 = PCA_retinatreatment$x[,3], PC_retinatreatment4 = PCA_retinatreatment$x[,4], 
                     tissue = colData(vst_retinatreatment)$Tissue,
                     treatment=colData(vst_retinatreatment)$Treatment,
                     genotype = colData(vst_retinatreatment)$Genotype)

ggplot(dataGG, aes(PC_retinatreatment1, PC_retinatreatment2, color=treatment, shape = genotype)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + labs(title = "Retina Treatment PCA 1/2")

ggplot(dataGG, aes(PC_retinatreatment3, PC_retinatreatment4, color=treatment, shape = genotype)) +
  geom_point(size=3) +
  xlab(paste0("PC3: ",percentVar[3],"% variance")) +
  ylab(paste0("PC4: ",percentVar[4],"% variance")) + labs(title = "Retina Treatment PCA 3/4")

```


## go enrichment
```{r}
# https://yulab-smu.top/biomedical-knowledge-mining-book/index.html
library(clusterProfiler)
library(enrichplot)
library(org.Mm.eg.db)

diff_genes <- res_treatment %>% 
  as_tibble(rownames = 'Gene') %>% 
  filter(padj < 0.01)


eg_diff_genes <- bitr(diff_genes$Gene, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")
eg_diff_genes <- diff_genes %>%
  left_join(., eg_diff_genes, by = c('Gene' = 'SYMBOL'))

# only keep genes that have a mean of >= 10 reads across all sample
expressed <- txi$counts[txi$counts %>% rowMeans() > 10,] %>% row.names()
eg_universe = bitr(res_treatment %>% 
                              as_tibble(rownames = 'Gene') %>% 
                              filter(Gene %in% expressed) %>% 
                              pull(Gene), fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")

ego <- enrichGO(gene= eg_diff_genes$ENTREZID,
                universe      = eg_universe$ENTREZID,
                OrgDb         = org.Mm.eg.db,
                ont           = "all",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                readable      = TRUE)
ego
```


```{r}
p1 <- dotplot(ego, showCategory=30) + ggtitle("Dotplot for GO")
p1
```

### Table
```{r}
ego <- setReadable(ego, OrgDb = org.Mm.eg.db, keyType = 'ENTREZID')
ego@result %>% DT::datatable()
```

### "CNET" Plot
Relationships between related GO terms with shared genes
```{r, fig.width=6, fig.height=4}
geneList <- eg_diff_genes$log2FoldChange
names(geneList) <- eg_diff_genes$Gene
cnet <- cnetplot(ego, foldChange = geneList) + scale_color_viridis_c(name = 'log2(FoldChange)')
cnet
```

### Enriched GO Term Relationships
Shows the "sets" of GO terms to increase understanding of what the diff gene sets are doing. There are many related GO terms.
```{r, fig.width=8, fig.height=8}
ego_pair <- pairwise_termsim(ego) 
go_term_relationships <- emapplot(ego_pair,layout="kk") #+ ggtitle("Enriched GO Term Relationships")
go_term_relationships
```
# session info
```{r}
sessionInfo()
```


