---
title: "Retina Untreated ABCA4KO vs WT Diff Test"
author: "Maya Bhattiprolu, David McGaughey"
date: "6/1/23"
output: 
  html_notebook:
    toc: TRUE
    toc_float: TRUE
---
```{r}
library(tidyverse)
library(tximport)
library(DESeq2)

list.files(path = "/Users/bhattiprolum2/Desktop/MitraMouse/")
afile = tximport('/Users/bhattiprolum2/Desktop/MitraMouse/MF_10M/quant.sf', type = "salmon", txOut = TRUE)

#install.packages('data.table')
files <- list.files(path ='/Users/bhattiprolum2/Desktop/MitraMouse/', recursive = TRUE, pattern = 'quant.sf', full.names = TRUE)

anno <- read_tsv(files[1])
anno$Gene <-sapply(anno$Name,function(x)strsplit(x,"\\|")[[1]][6])
anno_tximport <-anno %>%
  dplyr::select(target_id = Name, Gene)

txi <- tximport(files, type = "salmon", tx2gene = anno_tximport)
txi.dseq2 <- data.frame(txi$counts)

txi.dseq2
samples <- str_extract(files, "MF_\\d+M")
colnames(txi.dseq2) <- samples
colnames(txi.dseq2)

mouse <- read.csv('/Users/bhattiprolum2/Desktop/mitra_mouse.csv')

colnames(mouse)[1] <- "Sample"

mouse_metadata <- colnames(txi.dseq2) %>% 
  enframe() %>% 
  left_join(., mouse %>% mutate(value = gsub("-","_",  Sample), by = c("value"))) %>%  
  separate(Sample.description, c("Genotype","Treatment"), sep = '- ') %>% 
  mutate(Treatment = tolower(Treatment), 
                Treatment=trimws(Treatment))
dds <- DESeqDataSetFromTximport(txi, mouse_metadata, design = ~Tissue)
DESeq2Table <- DESeq(dds, parallel = T)
vst <- varianceStabilizingTransformation(DESeq2Table)
colnames(vst) <- colData(dds)$value

mouse_metadata2 <- mouse_metadata %>%
  dplyr::select(Sample, Treatment, Genotype, Tissue) %>% 
  mutate(MisLabel = case_when(Sample == 'MF-10M' ~ "Retina?",
                              Sample == 'MF-6M' ~ "Retina?"))
mouse_metadata2
```



# diff test
retina untreated ABCA4KO vs WT untreated
```{r}
dds_against_genotype <- DESeqDataSetFromTximport(txi, mouse_metadata, design = ~Genotype)
colnames(dds_against_genotype) <- colData(dds)$value
dds_retina_nodrug <- dds_against_genotype[,c('MF_11M', 'MF_1M', 'MF_3M', 'MF_9M')]

# may need to add this line:
# dds$condition <- droplevels(dds$condition)
dds_retina_nodrug$Genotype <- droplevels(dds_retina_nodrug$Genotype)

# start here
# dds <- DESeq(dds)
dds_retina_nodrug <- DESeq(dds_retina_nodrug)
res <- results(dds_retina_nodrug)
res

resLFC <- lfcShrink(dds_retina_nodrug, coef = "Genotype_WT_vs_ABCA4KO", type = "normal")

resLFC
res_tibble <- resLFC %>% as_tibble(rownames = "Gene") 

#sort by:
res_tibble %>%
  arrange(desc(log2FoldChange))

res_tibble %>%
  arrange(pvalue)

res_tibble %>%
  arrange(padj)

#Genes: Abca4, Zgrf1, Capza1, Fdxr, Leng8, Arhgap, Alpk1, C4b, Hk2, Ide

# pvalues
sum(res_tibble$padj <0.01, na.rm=TRUE)
res05 <- results(dds_retina_nodrug, alpha=0.05)
summary(res05)
sum(res05$padj < 0.05, na.rm=TRUE)
```
## ma-plot
```{r}
# stop here
# plotMA(resLFC, ylim=c(-2,2))
plotMA(res, ylim=c(-2,2))
plotMA(resLFC, ylim=c(-2,2))


# hint the "res" output needs to be turned into a tibble like so
# res_tibble <- res %>% as_tibble(rownames = "Gene") and then you can arrange by padj or log2FoldChange

```

## volcano plot
```{r}
volcano_data <- res %>%
  as_tibble(rownames = "Gene") %>%
  filter(!is.na(log2FoldChange)) 

volcano_data_filter_red <-  volcano_data %>% 
  filter(log2FoldChange > 1 |
           log2FoldChange < -1) %>% 
  filter(padj < 0.01,
         log1p(baseMean) > 5)

volcano_data_filter2 <- volcano_data %>%
  arrange(padj) %>%
  head(2)
#geom_label_repel(data = volcano_data_filter2)

volcano_data_filter3 <- volcano_data %>%
  arrange(padj) %>%
  filter(log2FoldChange >1) %>%
  filter(padj < 0.01, log1p(baseMean) > 5) %>%
  head(5)

volcano_data_filter4 <- volcano_data %>%
  arrange(padj) %>%
  filter(log2FoldChange < -1) %>%
  filter(padj < 0.01, log1p(baseMean) >5) %>%
  head(5) 

# manually select filter for baseMean by hand evaluating 
# distribution of log2(basemean)
# volcano_data %>% 
#     filter(log2FoldChange > 1 |
#                log2FoldChange < -1) %>% 
#     filter(padj < 0.01) %>% pull(baseMean) %>% summary()
# volcano_data %>% 
#     filter(log2FoldChange > 1 |
#                log2FoldChange < -1) %>% 
#     filter(padj < 0.01) %>% pull(baseMean) %>% log1p() %>% density() %>% plot()

library(ggrepel)
volcano_data %>% 
  ggplot(aes(x=log2FoldChange, y=-log10(pvalue), label = Gene)) + 
  geom_point() +
  geom_point(data = volcano_data_filter_red,
             color = 'red') + 
  geom_label_repel(data = volcano_data_filter3)+
   geom_label_repel(data = volcano_data_filter4)
```

## genes
```{r}
retina_counts <- as.data.frame(assay(dds_retina_nodrug))

retina_counts %>%
  as_tibble(rownames = 'Gene') %>%
  filter(Gene %in% c('Abca4', 'Arhgap29', 'Zgrf1', 'Capza1', 'Ide', 'Fdxr', 'Leng8', 'Alpk1', 'C4b', 'Hk2')) %>%
  pivot_longer(-Gene) %>%
  left_join(mouse_metadata %>% dplyr::select(name, value, Genotype, Treatment, Tissue), by = c('name' = 'value')) %>%
  ggplot(aes(x = Genotype, y = log1p(value), color = name)) +
  geom_boxplot(color = 'black') +
  geom_point(size = 4) +
  facet_wrap(~Gene, scales = 'free') + 
  labs( y = 'log counts')
```

## go enrichment
```{r}
library(clusterProfiler)
library(enrichplot)
library(org.Mm.eg.db)


diff_genes <- resLFC %>% 
  as_tibble(rownames = 'Gene') %>% 
  filter(padj < 0.01) %>%
  filter(abs(log2FoldChange) > 1)


eg_diff_genes <- bitr(diff_genes$Gene, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")
eg_diff_genes <- diff_genes %>%
  left_join(., eg_diff_genes, by = c('Gene' = 'SYMBOL'))

# only keep genes that have a mean of >= 10 reads across all sample

expressed <- txi$counts[txi$counts %>% rowMeans() > 10,] %>% row.names()
eg_universe = bitr(resLFC %>% 
                              as_tibble(rownames = 'Gene') %>% 
                              filter(Gene %in% expressed) %>% 
                              pull(Gene), fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")

ego <- enrichGO(gene= eg_diff_genes$ENTREZID,
                universe      = eg_universe$ENTREZID,
                OrgDb         = org.Mm.eg.db,
                ont           = "all",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
                readable      = TRUE)
ego
```

### table
```{r}
ego <- setReadable(ego, OrgDb = org.Mm.eg.db, keyType = 'ENTREZID')
ego@result %>% DT::datatable()
```

### cnet plot
```{r, fig.width=6, fig.height=4}
geneList <- eg_diff_genes$log2FoldChange
names(geneList) <- eg_diff_genes$Gene
cnet <- cnetplot(ego, foldChange = geneList) + scale_color_viridis_c(name = 'log2(FoldChange)')
cnet
```

### Enriched GO Term Relationships
```{r, fig.width=16, fig.height=16}
ego_pair <- pairwise_termsim(ego) 
go_term_relationships <- emapplot(ego_pair,layout="kk") #+ ggtitle("Enriched GO Term Relationships")
go_term_relationships
```
# session info
```{r}
sessionInfo()
```

